### DISCUSSION TITLE
Understanding how table captions work when using Typst format

### DISCUSSION THREAD

**Discussion by mattwarkentin** [quarto-dev/quarto-cli#11456] [Q&A] [answered]
### Description

Hi,

I have read https://github.com/quarto-dev/quarto-cli/issues/9369 and some of the off-shooting discussions. I am still not clear on why table captions ONLY show up when you add a label and the label must have the `tbl-` prefix. This seems inconsistent with figures whereby just adding `#| fig-cap: This is a figure caption` works as expected.

For example:
````
---
title: "Untitled"
format: typst
---

```{r}
#| tbl-cap: This is my caption but it does not show up
knitr::kable(iris[1:5,])
```

```{r}
#| label: foo
#| tbl-cap: This is my caption but it does not show up
knitr::kable(iris[1:5,])
```

```{r}
#| label: tbl-foo
#| tbl-cap: This caption works
knitr::kable(iris[1:5,])
```
````

**Comment by cscheid** [comment, marked as answer]
The unsatisfactory answer is that "Figures are special", which is a historical accident from Pandoc's handling of Figure captions. Captions in figures without labels are best thought of as "Pandoc captions", not "Quarto captions".

In Quarto, captions are associated with "floats" (more accurately, with `FloatRefTarget` objects). These objects can be Figure, Table, or [most anything else](https://quarto.org/docs/authoring/cross-references-custom.html). Floats need to have ids whose prefix `fig-`, `tbl-`, etc associate them with the appropriate type (Figure, Table, etc).

**Reply by mattwarkentin** [reply]
I think I'm kind of following what you're saying, without really understanding the technical details of how this is all working. So, is the expectation that in order to get table captions in Typst one must manually label their code blocks that produce tables?

**Reply by cscheid** [reply]
If they're coming from code blocks, then yes, it is. Alternatively, you can have your code block emit only the table information itself, and use the [div float syntax](https://quarto.org/docs/authoring/cross-references-divs.html):

````
---
format: typst
---

::: {#tbl-example}

```{r}
#| echo: false
gt::gt(gt::exibble)
```

A caption.
:::

See @tbl-example.
````

Apologies for the badly-formatted table, but it gets the point across.

<img width="705" alt="image" src="https://github.com/user-attachments/assets/12bb601b-0693-4cb4-9520-1bb79728d2ec">

**Reply by mattwarkentin** [reply]
Okay, understood. Thanks for the reply!
